pkg/healthcheck/healthcheck.go

```diff
@@ -1463,6 +1546,169 @@ func FetchLinkerdConfigMap(k kubernetes.Interface, controlPlaneNamespace string)
+		actualStr := strings.Join(actual, ",")
+
+		if expectedStr != actualStr {
+			return fmt.Errorf("expected %s, got %s", expectedStr, actualStr)
+		}
+
+		return nil
+	}
+
+	cr, err := hc.kubeAPI.RbacV1().ClusterRoles().Get(linkerdServiceMirrorComponentName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Could not obtain service mirror ClusterRole: %s", err)
+	}
+
+	var policyRule *v1.PolicyRule
+
+	for _, r := range cr.Rules {
+		if len(r.APIGroups) == 1 && r.APIGroups[0] == "" {
+			rule := r
+			policyRule = &rule
+		}
+	}
+
+	if policyRule == nil {
+		return fmt.Errorf("Service mirror ClusterRole is missing expected policy rule")
+	}
+
+	if err := compare(expectedServiceMirrorPolicyVerbs, policyRule.Verbs); err != nil {
+		return fmt.Errorf("Service mirror ClusterRole is missing verbs: %s", err)
+	}
+
+	if err := compare(expectedServiceMirrorPolicyResources, policyRule.Resources); err != nil {
+		return fmt.Errorf("Service mirror ClusterRole is missing required resources: %s", err)
+	}
+
+	return nil
+}
+
+func (hc *HealthChecker) checkRemoteClusterConnectivity() error {
+	options := metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("%s=%s", "type", k8s.MirrorSecretType),
+	}
+	secrets, err := hc.kubeAPI.CoreV1().Secrets(corev1.NamespaceAll).List(options)
+	if err != nil {
+		return err
+	}
+
+	if len(secrets.Items) == 0 {
+		return &SkipError{Reason: "no remote cluster configs"}
+	}
+
+	var errors []string
+	for _, s := range secrets.Items {
+		secret := s
+		config, err := sm.ParseRemoteClusterSecret(&secret)
+		if err != nil {
+			errors = append(errors, fmt.Sprintf("*  secret: [%s/%s]: could not parse config secret: %s", secret.Namespace, secret.Name, err))
+			continue
+		}
+
+		clientConfig, err := clientcmd.RESTConfigFromKubeConfig(config.APIConfig)
+		if err != nil {
+			errors = append(errors, fmt.Sprintf("* secret: [%s/%s] cluster: [%s]: unable to parse api config: %s", secret.Namespace, secret.Name, config.ClusterName, err))
+			continue
+		}
+
+		remoteAPI, err := k8s.NewAPIForConfig(clientConfig, "", []string{}, requestTimeout)
+		if err != nil {
+			errors = append(errors, fmt.Sprintf("* secret: [%s/%s] cluster: [%s]: could not instantiate remote api: %s", secret.Namespace, secret.Name, config.ClusterName, err))
+			continue
+		}
+
+		canGet := false
+		canList := false
+		canWatch := false
+		if err := hc.checkCanPerformAction(remoteAPI, "get", corev1.NamespaceAll, "", "v1", "services"); err == nil {
+			canGet = true
+		}
+
+		if err := hc.checkCanPerformAction(remoteAPI, "list", corev1.NamespaceAll, "", "v1", "services"); err == nil {
+			canList = true
+		}
+
+		if err := hc.checkCanPerformAction(remoteAPI, "watch", corev1.NamespaceAll, "", "v1", "services"); err == nil {
+			canWatch = true
+		}
+
+		if !canGet || !canList || !canWatch {
+			errors = append(errors, fmt.Sprintf("* cluster: [%s]: Insufficient permissions GET: %v, LIST: %v, WATCH: %v", config.ClusterName, canGet, canList, canWatch))
```

Comment:
Nit: could you unify the error format with the one from `checkServiceMirrorLocalRBAC()`? (there you have something like `Service mirror ClusterRole is missing verbs: expected create,delete,get,list,update,watch, got create,delete,get,list,watch`